cmd/publishing-bot/config/rules.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Coordinate of a piece of code
+type Coordinate struct {
+	Repository string `yaml:"repository"`
+	Branch     string `yaml:"branch"`
+	// Dir from repo root
+	Dir string `yaml:"dir,omitempty"`
+}
+
+func (c Coordinate) String() string {
+	return fmt.Sprintf("[repository %s, branch %s, subdir %s]", c.Repository, c.Branch, c.Dir)
+}
+
+type BranchRule struct {
+	Source      Coordinate `yaml:"source"`
+	Destination Coordinate `yaml:"destination"`
+	// k8s.io/* repos the dst dependes on
+	Dependencies []Coordinate `yaml:"dependencies,omitempty"`
```

Comment:
replace Coordinate with `type Dependency struct { repository, branch string }`

Reply:
@sttts the dir is needed for Source (used as a 'subdir')